src/components/Utils/Validate.js

```diff
@@ -4,8 +4,10 @@ const Required = (value) => {
   return 'Required!';
 };
 
-const RequiredInput = () => {
-  return false;
+const RequiredRefNumber = (value, initialValues) => {
+  const refNumber = initialValues.vendor_detail.ref_number;
+  if (refNumber && !value) return 'Required!';
```

Comment:
make `'Required!'` a constant, it could be changed in future, so we don't need to change it over all usages.